pywbemtools/pywbemcli/_common.py

```diff
@@ -655,14 +655,17 @@ def create_params(classname, cim_method, kv_params):
             params.append((name, cim_value))
         return params
 
+    assert isinstance(objectname, (CIMClassName, CIMInstanceName))
     conn = context.pywbem_server.conn
-    classname = objectname.classname \
-        if isinstance(objectname, (CIMClassName, CIMInstanceName)) \
-        else objectname
+    classname = objectname.classname
 
+    cmd_namespace = options['namespace']
     cim_class = conn.GetClass(
         classname,
-        namespace=options['namespace'], LocalOnly=False)
+        namespace=cmd_namespace, LocalOnly=False)
+
+    if options['namespace']:
+        objectname.namespace = cmd_namespace
```

Comment:
This modifies the object that is passed in as an argument. That is a side effect I think we should avoid if possible, even if it is just an internal method.

In the instance-level invoke method, a get_instancename() is invoked that returns a CIMInstance object with namespace set from the options, and that is passed to process_invokemethod(), so that one does not need to inject the namespace. Maybe we can do it similarly for the class-level invoke method, i.e. pass a CIMClassName with namespace set. We would still reduce the objectname parameter like you did, to no longer accept a string.

Reply:
Fixed. DONE.  Change all to avoid passing options to this method